lib/wallet/wallet_change_notifier.dart

```diff
@@ -1,11 +1,119 @@
-import 'package:flutter/material.dart';
+import 'package:f_logs/f_logs.dart';
+import 'package:flutter/material.dart' hide Flow;
+import 'package:ten_ten_one/bridge_generated/bridge_definitions.dart';
+import 'package:ten_ten_one/models/amount.model.dart';
+import 'package:ten_ten_one/models/payment.model.dart';
+
+import 'package:ten_ten_one/ffi.io.dart' if (dart.library.html) 'ffi.web.dart';
 
-/// Responsible for managing the state across the different Wallet screens.
 class WalletChangeNotifier extends ChangeNotifier {
+  WalletInfo walletInfo = WalletInfo(
+      balance: Balance(
+          onChain: OnChain(trustedPending: 0, untrustedPending: 0, confirmed: 0),
+          offChain: OffChain(available: 0, pendingClose: 0)),
+      bitcoinHistory: List.empty(),
+      lightningHistory: List.empty());
+
+  List<PaymentHistoryItem> history = List.empty();
+
   int selectedIndex = 0;
 
-  update(int index) {
+  set(int index) {
     selectedIndex = index;
     super.notifyListeners();
   }
+
+  Future<void> update(WalletInfo? walletInfo) async {
+    if (walletInfo == null) {
+      // skip empty wallet info update.
+      return;
+    }
```

Comment:
❓Why do we need to handle this? Why do we ever emit an empty event?

Reply:
because I am the `build_wallet_info` can fail and I am transforming that Result into an `Option`. I kind of liked that approach, as it really doesn't matter if now and then a `build_wallet_info` fails. However, currently the error would not get logged, which is a disadvantage.

Reply:
I do think we should log the error. I'm not sure exactly what component should be responsible for logging it though.

It might be useful to emit the error to the event stream so that the UI can display it. I do agree that we might want to allow some errors to occur before reporting anything to the user, but the best strategy isn't obvious to me.